lib/buffer.js

```diff
@@ -1208,13 +1209,9 @@ if (internalBinding('config').hasIntl) {
   };
 }
 
-let DOMException;
-
-const lazyInvalidCharError = hideStackFrames((message, name) => {
-  if (DOMException === undefined)
-    DOMException = internalBinding('messaging').DOMException;
-  throw new DOMException('Invalid character', 'InvalidCharacterError');
-});
+const invalidCharError = () => {
```

Comment:
You should keep lazy in the function name so it's clearer it's lazy loading the DOMException. Also, the `hideStackFrame` should stay, unless there's a reason to remove it?
```suggestion
const lazyInvalidCharError = hideStackFrames(() => {
```

Reply:
The called `lazyDOMException` function already hidden stack frames once. Should it be hidden once again?

Reply:
I'd think so, `hideStackFrames` works using function name: https://github.com/nodejs/node/blob/6dbff83730ecba32df3f734c474a02614ff6b8a6/lib/internal/errors.js#L374-L380
Here you are defining a new function that doesn't have the node internal prefix in its name, so it will appear on the stack trace.